components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -144,7 +157,63 @@ private void executeClientAuthenticators(HttpServletRequest request, OAuthClient
+                    clientId = extractClientId(oAuthClientAuthenticator, request,
+                            bodyContentMap);
+                    if (StringUtils.isNotBlank(clientId)) {
+                        break;
+                    }
+                } catch (OAuthClientAuthnException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Client ID cannot be extracted using the " + oAuthClientAuthenticator.getName());
+                    }
+                }
+            }
+            if (StringUtils.isNotBlank(clientId)) {
+                try {
+                    if (OAuth2Util.isFapiConformantApp(clientId)) {
+                        List<OAuthClientAuthenticator> authenticatorsForFapiApp = getAuthenticators(clientId, true);
+                        if (authenticatorsForFapiApp.isEmpty()) {
+                            setErrorToContext(OAuth2ErrorCodes.INVALID_REQUEST, "No valid authenticators found for " +
+                                    "the application.", oAuthClientAuthnContext);
+                        } else {
+                            authenticatorsForFapiApp.forEach(oAuthClientAuthenticator -> {
+                                executeAuthenticator(oAuthClientAuthenticator, oAuthClientAuthnContext, request,
+                                        bodyContentMap);
+                            });
+                        }
+                        return;
+                    } else {
+                        List<OAuthClientAuthenticator> authenticatorsForNonFapiApp = getAuthenticators(clientId, false);
+                        if (!authenticatorsForNonFapiApp.isEmpty()) {
+                            authenticatorsForNonFapiApp.forEach(oAuthClientAuthenticator -> {
+                                executeAuthenticator(oAuthClientAuthenticator, oAuthClientAuthnContext, request,
+                                        bodyContentMap);
+                            });
+                            return;
+                        }
+                    }
+                } catch (IdentityOAuth2ClientException e) {
+                    throw new OAuthClientAuthnException("Could not find an existing app for clientId: " + clientId,
+                            OAuth2ErrorCodes.INVALID_CLIENT);
+                } catch (IdentityOAuth2Exception e) {
+                    throw new OAuthClientAuthnException("Error while obtaining the service provider for client_id: " +
+                            clientId, OAuth2ErrorCodes.SERVER_ERROR);
+                }
+            } else {
+                setErrorToContext(OAuth2ErrorCodes.INVALID_CLIENT, "Client ID not found in the request.",
+                        oAuthClientAuthnContext);
+            }
+        } catch (IdentityOAuth2Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while processing the request to validate the client authentication method.");
+            }
```

Comment:
Lets make this a error log and also log the exception.

Reply:
Fixed in 51fdd762037ef49e2ffb96749e616e1a0343f202